src/cryptonote_core/blockchain.cpp

```diff
@@ -2981,6 +2982,48 @@ bool Blockchain::check_tx_inputs(transaction& tx, tx_verification_context &tvc,
       return false;
     }
   }
+  else if (tx.version == transaction::version_3_deregister_tx)
```

Comment:
Reject other versions of deregister TX's by checking the blockchain DB up to (VOTE_LIFETIME_BY_HEIGHT * 2) number of blocks from the height that the vote was submitted for.